lib/versioned/suite.js

```diff
@@ -29,23 +29,19 @@ util.inherits(Suite, EventEmitter)
 
 Suite.prototype.start = function(cb) {
   // Figure out the union of all packages required by all of the tests.
-  //
-  // TODO: When Node v0.8 and v0.10 are no longer supported change this to use `Set`.
-  var packages = Object.keys(this.testFolders.map(function(folder) {
-    var testPkg = require(path.join(folder, 'package'))
-    return testPkg.tests ? testPkg.tests.map(function(test) {
-      return Object.keys(test.dependencies)
-    }) : []
-  }).reduce(function(pkgs, b) {
-    b.forEach(function(t) {   // For each test `t` in bundle `b`
-      t.forEach(function(p) { // For each package `p` in test `t`
-        pkgs[p] = true        // Mark package `p` as found.
+  let packages = new Set()
+  this.testFolders.forEach((folder) => {
+    let testPackage = require(path.join(folder, 'package'))
+    if (testPackage.tests) {
+      testPackage.tests.forEach((test) => {
+        let dependencies = Object.keys(test.dependencies)
+        dependencies.forEach((dep) => packages.add(dep))
       })
-    })
-    return pkgs
-  }, {}))
+    }
+  })
+  packages = Array.from(packages)
```

Comment:
yay descriptive names.

another minor comment on swapping variable type. prob doesn't really matter given contained to same function.